### ISSUE TITLE
Modify code generator for graphql create and update input types to respect required fields in db model 

### ISSUE THREAD

**Issue by sebargarcia** [silentium-labs/merlin-gql#59]
Today for create and update input types MerlinGql assigns default values so this causes that when we generate the GraphQL schema, these required attributes don´t be required. I think that the generated GraphQL schema must honor these DB model constraints.

The problem occurs because we use strict mode in tsconfig. 

The solution is simple. Instead of generating a default value, set the attribute as required.

Example:

These generate today

```typescript
@InputType()
export class RoleProfileCreateInput extends BaseInputFields implements Partial<RoleProfile> {
    @Field((type) => Date)
    created: Date = new Date();

    @Field((type) => Date, { nullable: true })
    updated?: Date;

    @Field((type) => Date, { nullable: true })
    deleted?: Date;

    @Field((type) => Float)
    roleId: number = 0;

    @Field((type) => Float)
    profileId: number = 0;
}
```

it should generate

```typescript
@InputType()
export class RoleProfileCreateInput extends BaseInputFields implements Partial<RoleProfile> {
    @Field((type) => Date)
    created!: Date;

    @Field((type) => Date, { nullable: true })
    updated?: Date;

    @Field((type) => Date, { nullable: true })
    deleted?: Date;

    @Field((type) => Float)
    roleId!: number;

    @Field((type) => Float)
    profileId!: number;
}
```

edit: markup

**Comment by ezequielzacca** [comment]
You might have issues by doing that, remember what we discussed some months ago about how ¨reflection¨ works in javascript.
If you define a class like so:

```typescript
class MyClass {
    name: String!
    born: Date!
}
```

and you try to get the list of fields for an instance member of the class like so
```typescript
const member = new MyClass();
const properties = Object.keys(member);
```
What you will get is an empty array of properties, that is because in javascript, undefined properties technically don't exist within the object, so they don't get listed with Object.keys.
This is a known issue with javascript and that is why the strict mode is very usefull.

One way to solve that is to initialize the properties as null, null is not undefined so they will get listed by Object.keys, like so:
```typescript
class MyClass {
    name: String | null = null;
    born: Date | null = null;
}
```

**Comment by sebargarcia** [comment]
you are right I forgot that. I left this open for looking for an alternative

**Comment by ezequielzacca** [comment]
On a side note
```typescript
@InputType()
export class RoleProfileCreateInput extends BaseInputFields implements Partial<RoleProfile> {
    @Field((type) => Date)
    created: Date = new Date();
}
```

in this case, created **_IS REQUIRED_** by the GraphQL Schema. Unless it has nullable:true, it should be marked as required in the generated schema, if not then something is wrong. The `@Field` decorator metadata is not dependant on the javascript instance field having a value or not, because it forces GraphQL Clients to provide a value for the `created`field.
If you are talking about the **TypeORM** `@Column` validation, then yes, assigning a default value would make it so the validation passes due to the default value, but initializing the field as null instead of new Date() should fix that.

In short, we could modify the generator so that instead of a default value of new Date for dates, or an empty string for strings, or 0 for numbers, it assigns null as the default value for everything.
